car/car.go

```diff
@@ -44,12 +44,8 @@ func (c *Car) SeatbeltManipulator() parts.SeatbeltManipulator {
 
 func (c *Car) GearboxManipulator() parts.GearboxManipulator {
 	if c.gearbox == nil {
-		motor := c.MotorManipulator()
-		wheels := c.WheelManipulator()
-		return &parts.Gearbox{
-			Motor : motor,
-			Wheels: wheels,
-	}}
+		return c.gearbox.NewGearbox(c.motor, c.wheels)
```

Comment:
This should probably also use the MotorManipulator and WheelManipulator function, since that is a singleton pattern, and these may be nil at this point. @irving-luna

Reply:
Good catch. Fixing it